tensorflow/python/ops/string_ops.py

```diff
@@ -117,6 +118,7 @@ def string_split(source, delimiter=" "):  # pylint: disable=invalid-name
 ops.NotDifferentiable("ReduceJoin")
 ops.NotDifferentiable("StringJoin")
 ops.NotDifferentiable("StringSplit")
+ops.NotDifferentiable("Substr")
```

Comment:
This was added as I was cleaning up this PR, since most of the string ops are registered as non-differentiable. I'll remove it if you think it's unnecessary.